exampleplugins/5.2/SwagPaymentExample/Controllers/Frontend/PaymentExample.php

```diff
@@ -0,0 +1,102 @@
+     * Index action method.
+     *
+     * Forwards to the correct action.
+     */
+    public function indexAction()
+    {
+        /**
+         * Check if one of our payment methods is selected. Else return to default controller.
+         */
+        switch ($this->getPaymentShortName()) {
+            case 'example_payment_invoice':
+                return $this->redirect(['action' => 'gateway', 'forceSecure' => true]);
+            case 'example_paxment_cc':
+                return $this->redirect(['action' => 'direct', 'forceSecure' => true]);
+            default:
+                return $this->redirect(['controller' => 'checkout']);
+        }
+    }
+
+    /**
+     * Gateway action method.
+     *
+     * Collects the payment information and transmit it to the payment provider.
+     */
+    public function gatewayAction()
+    {
+        $this->View()->assign('gatewayUrl', 'www.example.com/gateway?' . $this->getUrlParameters());
+    }
+
+    /**
+     * Direct action method.
+     *
+     * Collects the payment information and transmit it to the payment provider.
+     */
+    public function directAction()
+    {
+        $this->redirect('www.example.com/direct?' . $this->getUrlParameters());
+    }
+
+    /**
+     * Return action method
+     *
+     * Reads the transactionResult and represents it for the customer.
+     */
+    public function returnAction()
+    {
+        $status = $this->Request()->getParam('status');
+        $hash = $this->Request()->getParam('hash');
+        $user = $this->getUser();
+        $billing = $user['billingaddress'];
+        $transactionId = $this->Request()->getParam('transactionId');
+
+        if ($status == 'accepted' && $hash == md5(implode('', [$this->getAmount(), $billing['customernumber']]))) {
+            $this->saveOrder($transactionId, $hash, self::PAYMENTSTATUSPAID);
+            $this->redirect(array('controller' => 'checkout', 'action' => 'finish', 'sUniqueID' => $hash));
+        } else {
+            $this->forward('cancel');
+        }
+    }
+    /**
+     * Cancel action method
+     */
+    public function cancelAction()
+    {
+
+    }
+
+    /**
+     * Creates the url parameters
+     */
+    private function getUrlParameters()
+    {
+        $router = $this->Front()->Router();
+        $user = $this->getUser();
+        $billing = $user['billingaddress'];
+
+        $parameter = [
+            'amount' => $this->getAmount(),
+            'customer' => [
+                'shopper_id' => $billing['customernumber'],
+                'addr_name' => $billing['firstname'] . ' ' . $billing['lastname'],
+            ],
+            'url' => [
+                'returnUrl' => $router->assemble(['action' => 'return', 'forceSecure' => true]),
+                'cancelUrl' => $router->assemble(['action' => 'cancel', 'forceSecure' => true]),
+            ],
+            'sessionId' => $this->container->get('session')->get('sessionId'),
+            'hash' => md5(implode('', [
```

Comment:
Sicher und sinnvoll ist dieser Hash so nicht. 
Du könntest z.B. $this->createPaymentUniqueId verwenden.